src/main/java/frc/robot/subsystems/Vision.java

```diff
@@ -1,31 +1,62 @@
 package frc.robot.subsystems;
 
+import edu.wpi.first.wpilibj.Relay;
 import edu.wpi.first.wpilibj.RobotBase;
+import edu.wpi.first.wpilibj.Relay.Value;
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 
-
 public class Vision extends SubsystemBase{
     private static final double[] EMPTY_TARGET_INFO = new double[] {0.0,0.0,0.0,0.0,0.0,0.0,0.0};
-    private double[] visionInfo;
+    private Relay spike;
 
     public Vision() {
-        visionInfo = EMPTY_TARGET_INFO;
+        spike = new Relay(0);
     }
 
     @Override
     public void periodic() {
-        // TODO Auto-generated method stub
-        
     }
 
-    public double[] getTargetInfo() {
-        if (RobotBase.isSimulation()) {
-            //TODO implement
-        } else {
-            visionInfo = SmartDashboard.getNumberArray("vision/target_info", EMPTY_TARGET_INFO);
+    // public double[] getTargetInfo() {
+    //     if (RobotBase.isSimulation()) {
+    //         //TODO implement
+    //     } else {
+    //         visionInfo = SmartDashboard.getNumberArray("vision/target_info", EMPTY_TARGET_INFO);
             
-        }        
-        return visionInfo;
+    //     }        
+    //     return visionInfo;
+    // }
+
+    // set vision processing mode, and set LED to match what is needed
+    public void setMode(String mode) {
+        setMode(mode, mode.equals("goalfinder"));
+    }
+
+    // set vision processing mode, and full control of LED
+    // should be used only for special cases
+    public void setMode(String mode, boolean led) {
+        SmartDashboard.putString("vision/active_mode/selected", mode);
+        setLedRing(led);
+    }
+    
+    public boolean getStatus() {
+        double[] visionData = SmartDashboard.getNumberArray("vision/target_info", EMPTY_TARGET_INFO);
+        // 1 = success, but it comes as a float, so test with a greater-than
+        return visionData[1] > 0.1;
+    }
+    
+    public double getDistance() {
+        double[] visionData = SmartDashboard.getNumberArray("vision/target_info", EMPTY_TARGET_INFO);
+        return visionData[3];
+    }
+    
+    public double getRobotAngle() {
+        double[] visionData = SmartDashboard.getNumberArray("vision/target_info", EMPTY_TARGET_INFO);
+        return visionData[4] * 180.0 / Math.PI;
+    }
+
+    public void setLedRing (boolean on) {
```

Comment:
We may want to make this private. I think we want to use the mode to determine if the LED should be set. But there might be some test condition that requires this.

Reply:
Yes, make this private.